build.jam

```diff
@@ -7,16 +7,16 @@ import project ;
 
 project /boost/optional
     : common-requirements
-        <source>/boost/assert//boost_assert
-        <source>/boost/config//boost_config
-        <source>/boost/core//boost_core
-        <source>/boost/detail//boost_detail
-        <source>/boost/move//boost_move
-        <source>/boost/predef//boost_predef
-        <source>/boost/static_assert//boost_static_assert
-        <source>/boost/throw_exception//boost_throw_exception
-        <source>/boost/type_traits//boost_type_traits
-        <source>/boost/utility//boost_utility
+        <library>/boost/assert//boost_assert
```

Comment:
Too many dependencies. It should be

    assert
    config
    core
    move
    static_assert
    throw_exception
    type_traits